### ISSUE TITLE
built in tests broken, possibly due to bad php-doc annotations

### ISSUE THREAD

**Issue by shafquatnz** [silverstripe-labs/silverstripe-behat-extension#224]
**Problem**

On a clean installation with minimal scaffolding, the following behat tests fail:

```
@navigate
Feature:
  As a website user
  I want to navigate the site

  Scenario: The first test
   Given a "Page" "FooBar" with "URL"="baz" and "Content"="This is the foo page"
   And the "Page" "FooBar" is published
   When I go to "baz"
   Then I should see "foo"

  Scenario: The second test
  Given a "Page" "FooBaz" has the following data
    | Title      | FooBaz                |
    | URLSegment | bar                   |
    | Content    | This is the baz page. |
  And the "Page" "FooBaz" is published
  When I go to "bar"
  Then I should see "baz"
```

**Initial diagnosis**

"The first test" fails because the library creates the page with the url segment "foobar" instead of "baz". Put in a breakpoint and you will see the wrong URLSegment value on the table SiteTree_Live in the tmp database.

"The second test" result in an error: “Fatal error: Cannot use positional argument after named argument”. I suspect the php-doc annotation is broken.

The code snippets supplied in the README and as commented examples in the code (e.g. vendor\silverstripe\behat-extension\src\Context\FixtureContext.php) are not working. Some of the other steps like "Checks the order of two texts" in vendor\silverstripe\behat-extension\src\Context\BasicContext.php are also broken.

**How to reproduce**

A "vanilla" repo with basic installation and behat scaffolding is up at https://github.com/GOVTNZ/vanilla

Clone the repo, recreate .env file, run composer install, run a dev/build, and then setup apache / virtualhost / docker as per your needs. With chromedriver running on the background, the tests are run with the following command:

` SS_BASE_URL="http://locahost" php ./vendor/bin/behat --config=behat-local.yml @app`

To get the basic site / behat working the following files were created / modified:

composer.json
behat-local.yml
app\tests\behat\features\navigate.feature
app\tests\behat\src\FeatureContext.php
app\tests\behat\src\FixtureContext.php

Thanks.

## Acceptance criteria
- The `Given a "Page" "FooBaz" has the following data` syntax work and can be used to populate fixtures with tabular data.
- The `Checks the order of two texts` syntax works and can be used in test.
 
## Notes
- We me night to change the signature of some of the methods to get them working. Since some of them seem to not work at all, it seems reasonable to break some APIs in this context
- We'll create a follow up card for reviewing the examples in the README

## Pull requests
- https://github.com/silverstripe/gha-generate-matrix/pull/57
- https://github.com/silverstripe/silverstripe-behat-extension/pull/226
- https://github.com/silverstripe/silverstripe-behat-extension/pull/242

**Comment by maxime-rainville** [comment]
The first test is broken because the `URL` parameter should be `URLSegment`.
```diff
- Given a "Page" "FooBar" with "URL"="baz" and "Content"="This is the foo page"
+ Given a "Page" "FooBar" with "URLSegment"="baz" and "Content"="This is the foo page"
```

The second test looks broken because `FixtureContext::stepCreateRecordWithTable()` seems to expect a different call signature.

This PR fixes it https://github.com/silverstripe/silverstripe-behat-extension/pull/226, but I'm not 100% sure if that's the best way.

**Comment by shafquatnz** [comment]
thanks @maxime-rainville 

Can you please fix "Checks the order of two texts" in vendor\silverstripe\behat-extension\src\Context\BasicContext.php which is also broken?

Also can you update the "Examples" in the file changing URL to URLSegment? Here's an [example](https://github.com/silverstripe/silverstripe-behat-extension/blob/65177937f15c06cbd27390631504fb410c2c9289/src/Context/FixtureContext.php#L247)

Would appreciate if the test suite is updated as we're wanting to upgrade to PHP8 before the EOL of PHP7.4

**Comment by maxime-rainville** [comment]
@shafquatnz Honestly, this a low priority for us. 

If you want to do PRs for those other issue, we would be happy to have a look at them.

**Comment by sabina-talipova** [comment]
All PRs have been merged. Close issue.

**Comment by emteknetnz** [comment]
This has been released as [4.11.1](https://github.com/silverstripe/silverstripe-behat-extension/releases/tag/4.11.1) and [5.0.2](https://github.com/silverstripe/silverstripe-behat-extension/releases/tag/5.0.2)

**Comment by emteknetnz** [comment]
We've done follow up releases with an additional merged pr to [4.11.2](https://github.com/silverstripe/silverstripe-behat-extension/releases/tag/4.11.2) and [5.0.3](https://github.com/silverstripe/silverstripe-behat-extension/releases/tag/5.0.3)
